packages/@aws-cdk/aws-appmesh/lib/shared-interfaces.ts

```diff
@@ -92,9 +92,65 @@ export interface VirtualNodeListener {
   readonly portMapping?: PortMapping;
 
   /**
-   * Array fo HealthCheckProps for the node(s)
+   * Health checking strategy upstream nodes should use when communicating with the listener
    *
    * @default - no healthcheck
    */
   readonly healthCheck?: HealthCheck;
 }
+
+/**
+ * All Properties for Envoy Access logs for mesh endpoints
+ */
+export interface AccessLogConfig {
+  /**
+   * Path to a file to write access logs to
+   *
+   * @default - no file based access logging
+   */
+  readonly filePath?: string;
```

Comment:
Here's my suggestion 🙂.

To make `bind()` in `AccessLog` more flexible, let's actually have `AccessLogConfig` have an (optional) property called `accessLog` with the type `CfnVirtualNode.AccessLogProperty`.

Yes, I know we say we don't use L1 properties in our L2s, but that only applies to _input_ properties. This is an _output_ property. Complicated, I know 🙂.

Then, in `VritualNode`, you can simply do:

```ts
    const accessLoggingConfig = props.accessLog?.bind(this);

    const node = new CfnVirtualNode(this, 'Resource', {
      // ...
        logging: {
          accessLog: accessLoggingConfig.accessLog,
       },
       // ...
```

This has the huge benefit that customers don't have to wait for you to implement new access logging types - they have an "escape hatch" of updating their L1 layer, and then extending `AccessLog` themselves.

Reply:
Would this pattern support Virtual Gateways as well? https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appmesh-virtualgateway-virtualgatewaylogging.html

Would you implement the `bind` method to also set a `CfnVirtualGateway.VirtualGatewayAccessLogProperty;` property on the `AccessLogConfig` as well?

Reply:
If that's your plan, then I would eventually shoot for `AccessLogConfig` to have both properties!

```ts
export interface AccessLogConfig {
  readonly virtualNodeLogging?: CfnVirtualNode.AccessLogProperty;

  readonly virtualGatewayLogging?: CfnVirtualGateway.VirtualGatewayAccessLogProperty;
}

export class AccessLog {
  public static fromFilePath(filePath: string): AccessLog {
    return new FilePathAccessLog(filePath);
  }

  public abstract bind(scope: Construct): AccessLogConfig;
}

class FilePathAccessLog extends AccessLog {
  public constructor(private readonly filePath: string) {
  }

  public bind(_scope: Construct): AccessLogConfig {
    return {
      virtualNodeLogging: {
         file { path: filePath }
     },
     virtualGatewayLogging: {
        file: { path: filePath },
     },
  };
}
```

And then both `VirtualNode` and `VirtualGateway` read their respective fields after calling `bind()`.

The only way I could see this backfiring is if you were 100% convinced you would not ever have any differences between `CfnVirtualNode.AccessLogProperty` and `CfnVirtualGateway.VirtualGatewayAccessLogProperty` (but if that's the case, I would question why would you have 2 separate properties for them in CloudFormation).